buildSrc/src/main/groovy/io/deephaven/tools/docker/CombinedDockerRunTask.groovy

```diff
@@ -0,0 +1,308 @@
+
+    @OutputDirectory
+    final DirectoryProperty outputDir = project.objects.directoryProperty()
+
+    @Inject
+    CombinedDockerRunTask(ObjectFactory objectFactory) {
+        hostConfig = objectFactory.newInstance(DockerCreateContainer.HostConfig, objectFactory)
+    }
+
+    @Override
+    void runRemoteCommand() {
+        String containerId;
+        try {
+            // Create the container
+            CreateContainerCmd createContainerCmd = dockerClient.createContainerCmd(imageId.get())
+
+            //inlined setContainerCommandConfig(containerCommand)
+            if (hostConfig.network.getOrNull()) {
+                createContainerCmd.hostConfig.withNetworkMode(hostConfig.network.get())
+            }
+            if (entrypoint.getOrNull()) {
+                createContainerCmd.withEntrypoint(entrypoint.get())
+            }
+
+            CreateContainerResponse container = createContainerCmd.exec()
+            logger.quiet "Created container with ID '$container.id'."
+            containerId = container.id
+
+            // If there is an entrypoint (else no need to start, we just copy out of existing one
+            if (entrypoint.getOrNull()) {
+                // Start the container and wait for finish
+                StartContainerCmd startContainerCmd = dockerClient.startContainerCmd(containerId)
+                startContainerCmd.exec()
+
+                WaitContainerCmd containerCommand = dockerClient.waitContainerCmd(containerId)
+
+                ResultCallback<WaitResponse> callback = containerCommand.start()
+                def exitCode = callback.awaitStatusCode(awaitStatusTimeout.get(), TimeUnit.SECONDS)
+                logger.quiet "Container exited with code ${exitCode}"
+
+                // If the task failed, write all logs
+                if (exitCode != 0) {
+                    LogContainerCmd logCommand = dockerClient.logContainerCmd(containerId)
+                    logCommand.withContainerId(containerId)
+                    logCommand.withStdErr(true)
+                    logCommand.withStdOut(true)
+
+                    logCommand.exec(createCallback())?.awaitCompletion()
+
+
+                    throw new GradleException("Command '${entrypoint.get()}' failed with status code ${exitCode}, check logs for details")
+                }
+            }
+
+            // Copy output to internal output directory
+            CopyArchiveFromContainerCmd containerCommand = dockerClient.copyArchiveFromContainerCmd(containerId, remotePath.get())
+            logger.quiet "Copying '${remotePath.get()}' from container with ID '${containerId}' to '${outputDir.get()}'."
+
+            InputStream tarStream
+            try {
+                tarStream = containerCommand.exec()
+
+                def hostDestination = outputDir.get().asFile
+
+                hostDestination.deleteDir()
+                copyFile(tarStream, hostDestination)
+            } finally {
+                tarStream?.close()
```

Comment:
I'm not familiar w/ this syntax. Does this mean "if not null"?

Reply:
Apparently yes. This code is all inlined from the various docker gradle plugin tasks.